.github/workflows/skip_duplicate_workflow_runs.yaml

```diff
@@ -0,0 +1,32 @@
+name: Skip Duplicate Workflow Runs
+
+on:
+  workflow_call:
+    inputs:
+      paths:
+        description: 'A JSON-array with path patterns'
+        default: '[]'
+        required: false
+        type: string
+    outputs:
+      should_skip:
+        description: "The output from the skip_duplicate_runs job"
+        value: ${{ jobs.skip_duplicate_runs.outputs.should_skip }}
+
+permissions:
+  contents: read
+
+jobs:
+  skip_duplicate_runs:
+    name: 'Skip Duplicate Runs'
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@v5.2.0
+        with:
+          paths_ignore: '["**.md"]'
+          paths: ${{ inputs.paths }}
+          concurrent_skipping: 'same_content_newer' # skip newer runs with same content
```

Comment:
I think this setting incorrectly marked this workflow run as green https://github.com/puma/puma/actions/runs/9386209788 because everything was skipped

Sounds like the author of `skip-duplicate-actions` isn't that satisfied with `concurrent_skipping`: https://github.com/fkirc/skip-duplicate-actions/issues/312#issuecomment-1380952795